Misc compiler warnings, mid October 2022 edition

mcatanzaro opened the pull request:
#### 5b0c545def4c2ea8e0fe6a463e41ee135fcd454c
<pre>
Misc compiler warnings, mid October 2022 edition
<a href="https://bugs.webkit.org/show_bug.cgi?id=246295">https://bugs.webkit.org/show_bug.cgi?id=246295</a>

Reviewed by Adrian Perez de Castro.

This contains the usual regular batch of boring trivial warning fixes.
Most of these changes fix -Wreturn-type warnings, which as usual is the
biggest offender. There is also one -Wunused-parameter and one
-Wredundant-move.

* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitDerived):
* Source/WebCore/css/StyleRule.cpp:
(WebCore::StyleRuleBase::visitDerived):
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::createVertexArray):
* Source/WebCore/platform/VideoPixelFormat.cpp:
(WebCore::convertVideoFramePixelFormat):
* Source/WebCore/platform/graphics/ColorBlending.cpp:
(WebCore::blend):

Canonical link: <a href="https://commits.webkit.org/255432@main">https://commits.webkit.org/255432@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/342e3c67d830a88c28a45564d1c2e41897529aa2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92389 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1618 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22981 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102156 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96390 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1614 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29997 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84818 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98313 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98050 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/1065 "Found 1 new test failure: fast/forms/ios/file-upload-panel-capture.html (failure)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78902 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28006 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82983 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82676 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/71101 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36413 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16618 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34180 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17788 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3767 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38051 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40408 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39954 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36938 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fbb8b47c3188394e3652d86996fc41faa2eb8604. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/5202

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3a02f0ed27d85dcef0b2fb07b809b5e7c890f0ff. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/5202

aperezdc commented on Source/WebCore/css/StyleRule.cpp:
```diff
@@ -99,11 +99,10 @@ template<typename Visitor> constexpr decltype(auto) StyleRuleBase::visitDerived(
-        RELEASE_ASSERT_NOT_REACHED();
-    case StyleRuleType::Unknown:
-        RELEASE_ASSERT_NOT_REACHED();
+    default:
+        break;
     }
+    RELEASE_ASSERT_NOT_REACHED();
```
Wouldn't it be better to _not_ have a `default:` case? With this modification when enum elements get added, the compiler will not let us know that there are cases not being handled. I prefer to list the cases that are supposed to be unreachable explicitly, all together:

```suggestion
    case StyleRuleType::Margin:
    case StyleRuleType::Unknown:
    case StyleTuleType:: ... :
        RELEASE_ASSERT_NOT_REACHED();
    }
```

mcatanzaro replied:
Sure.

aperezdc reviewed (commented):
Just one nit, the rest LGTM.

mcatanzaro commented:
> Just one nit, the rest LGTM.

OK now?

mcatanzaro committed:
Misc compiler warnings, mid October 2022 edition
https://bugs.webkit.org/show_bug.cgi?id=246295

Reviewed by Adrian Perez de Castro.

This contains the usual regular batch of boring trivial warning fixes.
Most of these changes fix -Wreturn-type warnings, which as usual is the
biggest offender. There is also one -Wunused-parameter and one
-Wredundant-move.

* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitDerived):
* Source/WebCore/css/StyleRule.cpp:
(WebCore::StyleRuleBase::visitDerived):
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::createVertexArray):
* Source/WebCore/platform/VideoPixelFormat.cpp:
(WebCore::convertVideoFramePixelFormat):
* Source/WebCore/platform/graphics/ColorBlending.cpp:
(WebCore::blend):

Canonical link: https://commits.webkit.org/255432@main

webkit-commit-queue commented:
Committed 255432@main (5b0c545def4c): <https://commits.webkit.org/255432@main>

Reviewed commits have been landed. Closing PR #5202 and removing active labels.
